### PR TITLE
CI: doctest also actions

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Take back testing of `setup_tools.py`

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-pytorch-test-full.yml [hunk 1]
```diff
@@ -121,11 +121,15 @@ jobs:
         pip list
       shell: bash
 
-    - name: DocTests
+    - name: DocTests PL
       if: ${{ (steps.skip.outputs.continue == '1') }}
       working-directory: ./src
       run: pytest pytorch_lightning --cov=pytorch_lightning
 
+    - name: DocTests actions
```
Comment (carmocca) [inline .github/workflows/ci-pytorch-test-full.yml:129]: This should be a separate job, instead of putting it in the pl job

Comment (Borda) [reply .github/workflows/ci-pytorch-test-full.yml:129]: Good point, then maybe add it to the pkg install as this module is used for package creation...

Comment (Borda) [reply .github/workflows/ci-pytorch-test-full.yml]: moved it to testing pkgs install

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-pkg-install.yml [hunk 2]
```diff
@@ -43,6 +43,10 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: DocTests actions
+        if: ${{ (steps.skip.outputs.continue == '1') }}
```
Comment (akihironitta) [inline .github/workflows/ci-pkg-install.yml:47]:
```suggestion
      - name: DocTests actions
```

**Review by akihironitta — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Comment by Borda** [comment]
@awaelchli @carmocca, do we have some other way how to test this deprecation?
https://github.com/Lightning-AI/lightning/blob/633d14e67a12d2df792295ea6e0f5231aa335f74/setup.cfg#L37
it seems that now we need to have installed PL otherwise, pytest fails...

File: setup.cfg [hunk 3]
```diff
@@ -34,7 +34,8 @@ markers =
     cloud:Run the cloud tests for example
 filterwarnings =
     # error out on our deprecation warnings - ensures the code and tests are kept up-to-date
-    error::pytorch_lightning.utilities.rank_zero.LightningDeprecationWarning
+    # fixme: this needs to have PL installed so we canot run just test on lighting_lite or lightning_app
+    # error::pytorch_lightning.utilities.rank_zero.LightningDeprecationWarning
```
Comment (carmocca) [inline setup.cfg:38]: Opened https://github.com/Lightning-AI/lightning/pull/14902

File: .github/workflows/ci-pkg-install.yml [hunk 4]
```diff
@@ -43,6 +43,12 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: DocTests actions
+        working-directory: .actions/
+        run: |
+          grep 'pytest>' requirements/pytorch/test.txt | xargs -0 pip install
```
Comment (carmocca) [inline .github/workflows/ci-pkg-install.yml:49]:
```suggestion
          grep 'pytest>' ../requirements/pytorch/test.txt | xargs -0 pip install
```

**Pull request merged**